### PR TITLE
Add the JSON appendix

### PR DESCRIPTION
**Pull request by AdaNebot**
Added the JSON serialisation appendix. This solves issue #4

### CODE REVIEW

File: appendix_json.tex [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     [
+        {"t": { “order”:[index,index,...], “order”:[index, index...], ... } },
+        {"s": { “order”:[index,index,...], “order”:[index, index...], ... } },
+        ...
+        {"t": { “order”:[index,index,...], “order”:[index, index...], ... } },
+        {"s": { “order”:[index,index,...], “order”:[index, index...], ... } }
+     ]
+\end{verbatim}
+
+If the spatial (resp. temporal) order of the last pair is lower than the spatial (resp. temporal) MOCORDER, it is suggested to add a last couple, possibly associated with empty index lists to explicitly specify the spatial and temporal MOCORDER.
```
Comment (tomdonaldson) [inline appendix_json.tex:41]: The use of `resp.` here reads a little awkwardly for me, I'm thinking slightly stronger wording is applicable.  Maybe something like:

```suggestion
If the spatial or temporal orders of the last "order":[index,index,...] pair is lower than the respective spatial or temporal MOCORDER, then add an additional pair at the highest order with with an empty index list.
```

That said, I realize that I'm not sure how to do this for an ST-MOC, where the space and time sections are interleaved.  If there is only one spatial MOCORDER and one temporal MOCORDER for an ST-MOC, then should the final highest-order empty index list be specified for each space/time couple, or just once at the end of the whole ST-MOC?  Or are there instead different MOCORDERs for each space/time couple?

The ASCII MOC example `t61/1 s29/0-2 t61/3 s28/0 t60/2 61/6 s29/2 5` appears to have the second spatial value only going to level 28.  Maybe we can show an example ST-MOC that uses this advice?

**Review by tomdonaldson — CHANGES_REQUESTED** [review]
Thanks for this section!  Overall it looks good.  I have one formatting comment, one wording comment, and one content question where I realize I don't know when to specify the highest-order in the ASCII or JSON ST-MOC.

File: appendix_json.tex (hunk 1 above)
Comment (pferniqu) [reply appendix_json.tex:41]: 1) totally agree with the rephrasing.
2) The use of the additional couple to specify the MOCORDERs of an STMOC is only necessary if the last couple does not itself use the maximum orders (same logic as for a simple MOC). In the case of the example, it was therefore not necessary. I propose to follow Tom's suggestion and modify the example to illustrate this point as follows:
   [ { "t":{ "61":[0]}, "s":{ "29":[0,1,2]},
      { "t":{ "61":[2]}, "s":{ "28":[0]} },
      { "t":{ "61":[]}, "s":{ "29":[]}  ]

File: appendix_json.tex [hunk 2]
```diff
@@ -0,0 +1,50 @@
+\section{JSON encoding}
+If it is required to write a MOC as an JSON string,
+it is suggested to use the following syntax:
+
+\par\noindent
+\begin{verbatim}
```
Comment (tomdonaldson) [inline appendix_json.tex:6]:
```suggestion
\begin{Verbatim}
```

File: appendix_json.tex [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   { “order”:[index,index,...], “order”:[index, index...], ... }
+\end{verbatim}
```
Comment (tomdonaldson) [inline appendix_json.tex:8]:
```suggestion
\end{Verbatim}
```

File: appendix_json.tex [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    {“1“:[1,2,4], “2“:[12,13,14,21,23,25], “8“:[]}
+\end{Verbatim}
+
+As with ASCII encoding, the differentiation of a spatial MOC from a temporal MOC could be done by prefixing the JSON MOC with an 's' or a 't' using a dedicated JSON hierarchy level. In the absence of this information, the nature of the MOC is determined by its context of use.
+
+\par\noindent
+\begin{Verbatim}[frame=single, xrightmargin=-3cm] 
+     {"t": { “order”:[index,index,...], “order”:[index, index...], ... } }
+  or {"s": { “order”:[index,index,...], “order”:[index, index...], ... } }
+\end{verbatim}
```
Comment (tomdonaldson) [inline appendix_json.tex:26]:
```suggestion
\end{Verbatim}
```

File: appendix_json.tex [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+\par\noindent
+\begin{Verbatim}[frame=single, xrightmargin=-3cm] 
+     [
+        {"t": { “order”:[index,index,...], “order”:[index, index...], ... } },
+        {"s": { “order”:[index,index,...], “order”:[index, index...], ... } },
+        ...
+        {"t": { “order”:[index,index,...], “order”:[index, index...], ... } },
+        {"s": { “order”:[index,index,...], “order”:[index, index...], ... } }
+     ]
+\end{verbatim}
```
Comment (tomdonaldson) [inline appendix_json.tex:39]:
```suggestion
\end{Verbatim}
```

**Review by tomdonaldson — CHANGES_REQUESTED** [review]
There were a few straggling `verbatim`s (without the capital V).  When I tried those locally the formatting looked good to me.  I would just commit those myself, but I didn't want to conflict in case @AdaNebot or @pferniqu were attempting the same thing.

**Pull request merged**